File: Selenium/Project/Activity9.java
package seleniumActivities;

import java.time.Duration;
import java.util.List;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.testng.annotations.AfterClass;
import org.testng.annotations.BeforeClass;
import org.testng.annotations.Test;

public class Activity9 {
    WebDriver driver;

    @BeforeClass
    public void setUp() {
        driver = new FirefoxDriver();
        driver.get("https://alchemy.hguy.co/crm/");
        WebElement userName = driver.findElement(By.id("user_name"));
        WebElement password = driver.findElement(By.id("username_password"));
        WebElement login = driver.findElement(By.id("bigbutton"));
        
        userName.sendKeys("admin");
        password.sendKeys("pa$$w0rd");
        login.click();
    }

    @Test
    public void readaPopUp() {
        WebElement salesButton = driver.findElement(By.xpath("//*[@id=\"grouptab_0\"]"));
        Actions actions = new Actions(driver);
        actions.moveToElement(salesButton).perform();

        WebDriverWait wait1 = new WebDriverWait(driver, Duration.ofSeconds(30));
        WebElement leadOption = wait1.until(ExpectedConditions.visibilityOfElementLocated(
            By.xpath("//*[@id=\"moduleTab_9_Leads\"]")));

        leadOption.click();

        WebDriverWait wait2 = new WebDriverWait(driver, Duration.ofSeconds(30));
        WebElement listTable = wait2.until(ExpectedConditions.visibilityOfElementLocated(
                By.cssSelector(".list")));

        List<WebElement> rows = listTable.findElements(By.tagName("tr"));
        System.out.println("Total rows (including headers): " + rows.size());

        int nameCount = 0;  // This will track how many valid names we have printed

        for (int i = 2; i < rows.size(); i++) {  // Start from the 3rd row (index 2)
            WebElement row = rows.get(i);

            // Check if the row contains <td> elements
            List<WebElement> cells = row.findElements(By.tagName("td"));

            // Skip rows that don't contain enough <td> elements (header or other irrelevant rows)
            if (cells.size() <= 7) {
                continue; // Skip rows with less than 8 cells
            }

            String name = cells.get(2).getText();  // Assuming name is in the 3rd column (index 2)
            String leadUserName = cells.get(7).getText();  // Assuming userName is in the 8th column (index 7)

            // Skip rows with empty or non-relevant names
            if (name.equals("(1 - 20 of 54)")) {
                continue; // Skip rows with empty or non-relevant names
            }

            // Print Name and UserName for the first 10 valid rows
            System.out.println("Name: " + name + ", UserName: " + leadUserName);
            nameCount++;

            // Stop after printing 10 valid names
            if (nameCount == 10) {
                break;
            }
        }
    }
    
    @AfterClass
    public void tearDown() {
    	driver.quit();
    }


}

File: Selenium/Project/Activity9.py
from selenium import webdriver
from selenium.webdriver.common.action_chains import ActionChains
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
from selenium.webdriver.common.by import By
with webdriver.Firefox() as driver:
    driver.get("http://alchemy.hguy.co/crm")

    userName = driver.find_element(By.ID,"user_name")
    userName.send_keys("admin")
    
    password = driver.find_element(By.ID,"username_password")
    password.send_keys("pa$$w0rd")
	
    login = driver.find_element(By.ID,"bigbutton")
    login.click()

    salesButton = driver.find_element(By.XPATH,"//*[@id=\"grouptab_0\"]")
    actions = ActionChains(driver)
    actions.move_to_element(salesButton).perform()

    wait = WebDriverWait(driver, 10)
    leadOption = wait.until(EC.visibility_of_element_located((By.XPATH, "//*[@id=\"moduleTab_9_Leads\"]")))
    leadOption.click()

    wait = WebDriverWait(driver, 30)
    accountsTable = wait.until(EC.visibility_of_element_located((By.CSS_SELECTOR, ".list")))
    rows = accountsTable.find_elements(By.TAG_NAME,"tr")
    print(f"Total rows (including headers): {len(rows)}")

    nameCount = 0
    
    for i in range(2, len(rows)):
        row = rows[i]

        cells = row.find_elements(By.TAG_NAME, "td")

        if len(cells) <=7:
            continue

        name = cells[2].text
        leadUserName = cells[7].text

        if name == "(1 - 20 of 54)":
            continue 

        print(f"Name: {name} and Username: {leadUserName}")
        nameCount+=1

        if nameCount == 10:
            break 

    driver.quit()




